### PR TITLE
Discussion refactors

### PR DESCRIPTION
**Pull request by hakubo**
Hey,

While working on something else I did couple of changes to discussion code.
Do they look good to you?
@bkoval @kvas-damian @rogatty

### CODE REVIEW

File: front/main/app/components/discussion-post-editor.js [hunk 1]
```diff
@@ -9,20 +9,23 @@ export default DiscussionEditorComponent.extend({
...
+	didInsertElement() {
+		this._super(...arguments);
+		this.initializeStickyState();
+	},
+
 	/**
 	 * Initialize onScroll binding for sticky logic
 	 * @returns {void}
 	 */
-	initializeStickyState: Ember.on('didInsertElement', function () {
```
Comment (rogatty) [inline front/main/app/components/discussion-post-editor.js]: Seems like this was a discouraged pattern http://notmessenger.com/proper-use-of-ember-on/

We probably should add information to the guidelines.

**Comment by rogatty** [comment]
LGTM

**Comment by bkoval** [comment]
LGTM

**Comment by kvas-damian** [comment]
LGTM

File: front/main/app/components/discussion-editor.js [hunk 2]
```diff
@@ -176,7 +176,7 @@ export default Ember.Component.extend(ViewportMixin, {
 	 * @returns {void}
 	 */
 	didInsertElement() {
-		this._super();
+		this._super(...arguments);
```
Comment (vforge) [inline front/main/app/components/discussion-editor.js]: Can you use rest parameters explicitly instead of using `arguments`?

Comment (hakubo) [inline front/main/app/components/discussion-editor.js]: From what I see this is encouraged way. 
We don't have guideline about that yet also

Comment (hakubo) [inline front/main/app/components/discussion-editor.js]: especially when you look at the output of babel:
https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=true&spec=true&code=function%20a%20()%20%7B%0A%20%20console.log(...arguments)%3B%0A%7D%0A%0Afunction%20b(...rest)%20%7B%0A%20%20console.log(rest)%3B%0A%7D

Comment (vforge) [inline front/main/app/components/discussion-editor.js]: Thanks for the link! We just need to remember about not doing that with arrow functions.

**Pull request merged**